testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -0,0 +1,520 @@
+        "charmm": 1
+    }
+
+    with tmpdir.as_cwd():
+        with DCDFile('test.dcd', 'w') as dcd:
+            dcd.write_header(**header)
+            with pytest.raises(IOError):
+                dcd.write_header(**header)
+
+
+def test_write_header_wrong_mode():
+    # an exception should be raised on any attempt to use
+    # _write_header with a DCDFile object in 'r' mode
+    with DCDFile(DCD) as dcd:
+        with pytest.raises(IOError):
+            dcd.write_header(
+                remarks='Crazy!',
+                natoms=22,
+                istart=12,
+                nsavc=10,
+                delta=0.02,
+                charmm=1)
+
+
+def test_write_mode():
+    # ensure that writing of DCD files only occurs with properly
+    # opened files
+    with DCDFile(DCD) as dcd:
+        with pytest.raises(IOError):
+            dcd.write(xyz=np.zeros((3, 3)), box=np.zeros(6, dtype=np.float64))
+
+
+def write_dcd(in_name, out_name, remarks='testing', header=None):
+    with DCDFile(in_name) as f_in, DCDFile(out_name, 'w') as f_out:
+        if header is None:
+            header = f_in.header
+        f_out.write_header(**header)
+        for frame in f_in:
+            f_out.write(xyz=frame.xyz, box=frame.unitcell)
+
+
+@given(remarks=st.text(
+    alphabet=string.printable, min_size=0,
+    max_size=240))  # handle the printable ASCII strings
+@example(remarks='')
+def test_written_remarks_property(remarks, tmpdir):
+    # property based testing for writing of a wide range of string
+    # values to REMARKS field
+    testfile = 'test.dcd'
+    with DCDFile(DCD) as dcd:
+        header = dcd.header
+
+    with tmpdir.as_cwd():
+        header['remarks'] = remarks
+        write_dcd(DCD, testfile, header=header)
+        expected_remarks = remarks[:240]
+        with DCDFile(testfile) as f:
+            assert f.header['remarks'] == expected_remarks
+
+
+@pytest.fixture(scope='session')
+def written_dcd(tmpdir_factory):
+    testfile = 'test.dcd'
+    with DCDFile(DCD) as dcd:
+        header = dcd.header
+    testfile = tmpdir_factory.mktemp('dcd').join('test.dcd')
+    testfile = str(testfile)
+    write_dcd(DCD, testfile)
+    Result = namedtuple("Result", "testfile, header, orgfile")
+    return Result(testfile, header, DCD)
+
+
+def test_written_header(written_dcd):
+    header = written_dcd.header
+    with DCDFile(written_dcd.testfile) as dcd:
+        # need to pop charmm header for now.
+        header.pop('charmm')
+        dcdheader = dcd.header
+        dcdheader.pop('charmm')
+        assert dcdheader == header
+
+
+def test_written_num_frames(written_dcd):
+    with DCDFile(written_dcd.testfile) as dcd, DCDFile(
+            written_dcd.orgfile) as other:
+        assert len(dcd) == len(other)
+
+
+def test_written_dcd_coordinate_data_shape(written_dcd):
+    with DCDFile(written_dcd.testfile) as dcd, DCDFile(
+            written_dcd.orgfile) as other:
+        for frame, other_frame in zip(dcd, other):
```

Comment:
MUST USE `six.zip` ... here and below

In python 2 this will give **incorrect tests**, even though your test will pass:

```python
In [1]: import MDAnalysis as mda; from MDAnalysisTests.datafiles import GRO, XTC, TRR
In [2]: u1, u2 = mda.Universe(GRO, XTC), mda.Universe(GRO, TRR)
In [3]: for ts1, ts2 in zip(u1.trajectory, u2.trajectory):
   ...:     print(ts1.frame, ts2.frame)
   ...:
(0, 9)
(0, 9)
(0, 9)
(0, 9)
(0, 9)
(0, 9)
(0, 9)
(0, 9)
(0, 9)
(0, 9)
```